### ISSUE TITLE
Catch DELETE/INSERT DATA queries that include non-ground triples

### ISSUE THREAD

**Issue by Spadon** [RubenVerborgh/SPARQL.js#31]
Hello,

As from the SPARQL 1.1 specification, DELETE/INSERT DATA queries only allow ground triples. However, the following (incorrect) query does not return any parsing error :

`DELETE DATA { ?s ?p ?o }`

Best,

Mehdi

**Comment by RubenVerborgh** [comment]
Interesting, this is not specified in [the grammar](https://www.w3.org/TR/sparql11-query/#rDeleteData).

The [SPARQL 1.1 Update specification](https://www.w3.org/TR/sparql11-update/) indeed mentions that triples need to be ground.
I also noted an [exception on blank nodes](https://www.w3.org/TR/sparql11-query/#grammarBNodes).

**Comment by Spadon** [comment]
Indeed, both [QuadPattern](https://www.w3.org/TR/sparql11-query/#rQuadPattern) and [QuadData](https://www.w3.org/TR/sparql11-query/#rQuadData) refer to [Quads](https://www.w3.org/TR/sparql11-query/#rQuads) in the grammar.

**Comment by joachimvh** [comment]
Noticed this issue since I'm going through UPDATE stuff myself.

This actually is specified in the [grammar](https://www.w3.org/TR/sparql11-query/#sparqlGrammar), it's note 8 in the notes above the grammar table. 
The exception on blank nodes is note 10 (which links to the section you linked).
